### PR TITLE
reset queued stats if disk queue corrupted and restarted

### PR DESCRIPTION
**Pull request by Kokan**
The disk buffer is re-created when it detects it became corrupt, but the queued message statistics is not set accordingly.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

File: modules/diskq/logqueue-disk-non-reliable.c
Comment (MrAnno) [inline modules/diskq/logqueue-disk-non-reliable.c:324]: I don't understand this commit. Could you please explain it?

My understanding is that `_pop_head()` should decrement the queued messages counter whenever a message is returned from this function.

AFAIK `_maybe_move_messages_among_queue_segments()` is a transparent side effect, and it should not change the way we calculate queued messages.

File: modules/diskq/logqueue-disk-non-reliable.c
Comment (MrAnno) [inline modules/diskq/logqueue-disk-non-reliable.c:184]: This will eventually be FALSE when we can't fully saturate `qout`.

Comment (MrAnno) [reply modules/diskq/logqueue-disk-non-reliable.c:184]: Found it, sorry:
```
if (qdisk_get_length(self->super.qdisk) <= 0)
```

File: modules/diskq/logqueue-disk-non-reliable.c
Comment (MrAnno) [reply modules/diskq/logqueue-disk-non-reliable.c:324]: I get it now: the move operation can really fail, hence the conditional stats update.

**Review by MrAnno — DISMISSED** [review]
👍🏻

**Comment by Kokan** [comment]
@kira-syslogng please retest this

**Comment by Kokan** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by mitzkia** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
Oh, I forgot about the NEWS file entry. Could you please add one?

**Comment by Kokan** [comment]
Sure, you are on fire with news entries ;)

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
Sorry, I had no intention of being intrusive 😄

**Review by MrAnno — APPROVED** [review]

**Comment by gaborznagy** [comment]
I've checked if `memory_usage` has to be reset as well, but as I see, it will be adjusted during `restart` by `LogQueueDisk::start`.
Can you confirm, if I'm right, please?

File: modules/diskq/logqueue-disk-non-reliable.c
Comment (gaborznagy) [reply modules/diskq/logqueue-disk-non-reliable.c:333]: Of course, then this would be done under locking, which wanted to avoid.
it's just an idea.

Comment (Kokan) [reply modules/diskq/logqueue-disk-non-reliable.c:333]: I did consider this variation, this reduces the stress in the code here. I wanted to avoid that, due to it is easy to overlook the fact in `_maybe_move_messages_among_queue_segments` fatal stuff can happen. In general I would incline to do as you suggested, but here I think keeping the tension is helpful fur future debugging/review as this code raises the attention of the reader.

Comment (gaborznagy) [reply modules/diskq/logqueue-disk-non-reliable.c:333]: Thanks for the explanation. Good point about visibility, it did got my attention the first time.
I see your point, let's keep it this way then.

**Comment by gaborznagy** [comment]
@kira-syslogng retest this please;

**Review by gaborznagy — APPROVED** [review]
Waiting for a green Kira before merge.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**
